src/App.js

```diff
@@ -14,11 +15,96 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  componentDidUpdate() {
+    document.querySelectorAll('li').forEach((element) => {
+      if (this.state.selectedGood.includes(element.textContent)) {
+        element.classList.add('active');
+      } else {
+        element.classList.remove('active');
+      }
+    });
+
+    document.querySelectorAll('.clearAll-button').forEach((element) => {
+      if (this.state.selectedGood.length === 0) {
+        element.classList.add('hover');
+      } else {
+        element.classList.remove('hover');
+      }
+    });
+  }
+
+  selectedGoods = (selectedGood) => {
+    switch (selectedGood.length) {
+      case 0: return `No goods selected`;
+      case 1: return `${selectedGood} is selected`;
+      default: {
+        const lastGood = selectedGood[selectedGood.length - 1];
+
+        return `${selectedGood.slice(0, selectedGood.length - 1).join(', ')} 
+          and ${lastGood} are selected`;
+      }
+    }
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <>
+          <h1 className="title">
+            {this.selectedGoods(selectedGood)}
+          </h1>
+          <button
+            title="Clear the list of selected itemss"
+            type="button"
+            className="clearAll-button"
+            onClick={() => {
+              this.setState(() => ({
+                selectedGood: [],
+              }));
```

Comment:
```suggestion
              this.setState({ selectedGood: [] });
```
You can use simple setState (not functional) here because there's no need in previous state values